backend/api/Services/RobotService.cs

```diff
@@ -173,6 +173,12 @@ private IQueryable<Robot> GetRobotsWithSubModels()
                 .Include(r => r.VideoStreams)
                 .Include(r => r.Model)
                 .Include(r => r.CurrentArea)
+                .ThenInclude(r => r != null ? r.Deck : null)
```

Comment:
Use something more descriptive than r
```suggestion
                .ThenInclude(currentArea => currentArea != null ? currentArea.Deck : null)
```